test/matrix_generator.cc

```diff
@@ -1740,13 +1796,16 @@ void generate_matrix(
     lapack::laset( lapack::MatrixType::General, Sigma.size(), 1,
                    nan, nan, Sigma.data(), Sigma.size() );
 
+    char msg[ 256 ];
     TestMatrixType type;
     TestMatrixDist dist;
     real_t cond;
     real_t condD;
     real_t sigma_max;
     bool dominant;
-    decode_matrix<scalar_t>(params, A, type, dist, cond, condD, sigma_max, dominant);
+    int64_t zero_col;
```

Comment:
Should this function error when the user requests a zero column since it's not implemented?

Reply:
Either that or I should just implement it. Now what does zerocol mean for a Hermitian matrix... hmm.

Reply:
I added an implementation that sets row and col N to zero. For triangular and trapezoid matrices, setting row N to zero doesn't make too much sense, but there wasn't an obvious, simple way to handle those differently. Sample output:
```
pangolin slate/test> mpirun -np 4 ./tester --dim 10 --nb 3 --matrix rand_zerocol0.5 --check n --uplo l,u --verbose 4 henorm
% SLATE version 2023.08.25, id ce64cda5
% input: ./tester --dim 10 --nb 3 --matrix rand_zerocol0.5 --check n --uplo l,u --verbose 4 henorm
% 2023-09-27 00:10:48, 4 MPI ranks, CPU-only MPI, 4 OpenMP threads per MPI rank
                                                                                                            
type  origin  target   A  norm    uplo       n    nb    p    q      error   time (s)  ref time (s)  status  

% A: HermitianMatrix 10-by-10, 4-by-4 tiles, tileSize 3-by-3, uplo L
A_ = [
     0.6805        nan        nan            nan        nan        nan            nan        nan        nan            nan
     0.9900     0.7305        nan            nan        nan        nan            nan        nan        nan            nan
     0.1511     0.8275     0.0604            nan        nan        nan            nan        nan        nan            nan

     0.1603     0.9113     0.8815         0.2989        nan        nan            nan        nan        nan            nan
     0.         0.         0.             0.         0.            nan            nan        nan        nan            nan
     0.6592     0.8813     0.2548         0.6915     0.         0.6330            nan        nan        nan            nan

     0.5956     0.4351     0.0210         0.8936     0.         0.3624         0.1545        nan        nan            nan
     0.2267     0.9085     0.3559         0.2038     0.         0.6683         0.8483     0.9237        nan            nan
     0.1337     0.8756   0.005697         0.9543     0.         0.7100         0.7733     0.8352     0.8309            nan

     0.5879     0.6609     0.4310         0.0187     0.         0.9175         0.2005     0.9266     0.7034         0.7742
];
A = tril( A_ ) + tril( A_, -1 )';

   d    host    task   1     1   lower      10     3    2    2         NA   0.000189            NA  no check  

% A: HermitianMatrix 10-by-10, 4-by-4 tiles, tileSize 3-by-3, uplo U
A_ = [
     0.9588     0.3608     0.4184         0.7532     0.         0.6644         0.0343     0.0793     0.9325         0.7066
        nan     0.1976     0.6917         0.7610     0.         0.8699         0.5161     0.2728     0.8368         0.1435
        nan        nan     0.1051         0.1333     0.         0.1366         0.4302     0.8413     0.2068         0.9453

        nan        nan        nan         0.3759     0.         0.1042         0.0526     0.8515     0.3408         0.7051
        nan        nan        nan            nan     0.         0.             0.         0.         0.             0.    
        nan        nan        nan            nan        nan     0.8676         0.3734     0.2385     0.0233         0.6999

        nan        nan        nan            nan        nan        nan         0.5977     0.2495     0.4037         0.6288
        nan        nan        nan            nan        nan        nan            nan     0.0739     0.3419         0.3632
        nan        nan        nan            nan        nan        nan            nan        nan     0.6809         0.0860

        nan        nan        nan            nan        nan        nan            nan        nan        nan         0.0523
];
A = triu( A_ ) + triu( A_,  1 )';

   d    host    task   1     1   upper      10     3    2    2         NA   0.000146            NA  no check  

% Matrix kinds:
%  1: rand_zerocol0.5, cond = inf

% All tests passed: henorm
```
I also simplified some code that had nearly identical lower and upper cases.